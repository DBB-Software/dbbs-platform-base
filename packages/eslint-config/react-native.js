module.exports = {
  extends: ['@dbbs/eslint-config/react', 'plugin:react-native/all', '@react-native'],
  parser: '@typescript-eslint/parser',
  rules: {
    'comma-dangle': 'off',
    'react/react-in-jsx-scope': 'off'
  }
}
