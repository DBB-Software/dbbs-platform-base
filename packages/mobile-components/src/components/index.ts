export * from './screen'
export * from './icon'
export * from './smooth-slide-track'
