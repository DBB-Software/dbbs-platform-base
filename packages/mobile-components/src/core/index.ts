import { DefaultTheme } from 'react-native-paper'

export const appTheme = {
  ...DefaultTheme,
  // Specify custom property
  myOwnProperty: true,
  // Specify custom property in nested object
  colors: {
    ...DefaultTheme.colors
  }
}
