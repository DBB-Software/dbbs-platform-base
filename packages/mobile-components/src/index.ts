export * from 'react-native-paper'
export * from 'react-native-paper/react-navigation'

export * from './core'
export * from './hooks'
export * from './components'
export * from './svg'
export * from './utils'
