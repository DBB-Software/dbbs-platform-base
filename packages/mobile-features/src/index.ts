export * from './modules/remote-config'
export * from './modules/app-state'
