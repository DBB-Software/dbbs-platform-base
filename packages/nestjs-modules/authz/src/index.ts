export { AuthzModule } from './authz.module.js'
export { JwtStrategy } from './jwt.strategy.js'
