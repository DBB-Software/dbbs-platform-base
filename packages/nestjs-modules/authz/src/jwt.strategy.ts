import { Injectable } from '@nestjs/common'
import { PassportStrategy } from '@nestjs/passport'
import { ExtractJwt, Strategy } from 'passport-jwt'
import { passportJwtSecret } from 'jwks-rsa'
import { ConfigService } from '@nestjs/config'

@Injectable()
export class JwtStrategy extends PassportStrategy(Strategy) {
  constructor(private configService: ConfigService) {
    const audience = configService.get('AUTH0_AUDIENCE')
    const issuer = configService.get('AUTH0_ISSUER_URL')

    super({
      secretOrKeyProvider: passportJwtSecret({
        cache: true,
        rateLimit: true,
        jwksRequestsPerMinute: 5,
        jwksUri: `${issuer}.well-known/jwks.json`
      }),

      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
      audience,
      issuer,
      algorithms: ['RS256']
    })
  }

  validate(payload: unknown): unknown {
    return payload
  }
}
