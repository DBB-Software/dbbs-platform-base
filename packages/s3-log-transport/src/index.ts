import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3'

const DEFAULT_BATCH_SIZE = 5 // logs in a row
const DEFAULT_UPLOAD_INTERVAL_MS = 5000

export interface IS3TransportOptions {
  bucket: string
  folder: string
  batchSize?: number
  uploadInterval?: number
  region: string
}

function S3Transport(opts: IS3TransportOptions) {
  const { bucket, folder, region, batchSize = DEFAULT_BATCH_SIZE, uploadInterval = DEFAULT_UPLOAD_INTERVAL_MS } = opts
  const logBuffer: string[] = []
  const pmInstanceId = process.env.pm_id || 0

  const s3 = new S3Client({
    region
  })
  async function flush(size: number = logBuffer.length) {
    if (size > 0) {
      try {
        await s3.send(
          new PutObjectCommand({
            Bucket: bucket,
            Key: `${folder}/${new Date().toISOString()}{INSTANCE-${pmInstanceId}}.log`,
            Body: logBuffer.splice(0, size).join('\n')
          })
        )
      } catch (err) {
        console.error(err)
      }
    }
  }

  function initBatchTimer() {
    setInterval(() => {
      flush()
    }, uploadInterval)
  }

  function write(msg: string) {
    // TODO: make concurrent writing to buffer (#206)
    logBuffer.push(msg)

    if (logBuffer.length >= batchSize) {
      flush(batchSize)
    }
  }

  initBatchTimer()

  return { write }
}

export default S3Transport
