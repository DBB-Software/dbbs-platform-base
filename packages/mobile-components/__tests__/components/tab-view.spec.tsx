import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react-native'
import { Text } from 'react-native'
import { RenderTabBarProps, TabView } from '../../src'

describe('TabView Component', () => {
  const Tab1 = () => <Text>Tab 1 Content</Text>
  const Tab2 = () => <Text>Tab 2 Content</Text>

  const tabs = [
    { key: 'tab1', title: 'Tab 1', component: Tab1, testID: 'tab-1' },
    { key: 'tab2', title: 'Tab 2', component: Tab2, testID: 'tab-2' }
  ]

  it('renders correctly with default settings', () => {
    render(<TabView tabs={tabs} />)

    // Check if the initial tab is rendered
    expect(screen.getByText('Tab 1 Content')).toBeTruthy()
  })

  it('renders the correct content when the tab changes', async () => {
    render(<TabView tabs={tabs} />)

    // Initially, Tab 1 content should be visible
    expect(screen.getByText('Tab 1 Content')).toBeTruthy()

    // Change to the second tab
    fireEvent.press(screen.getByTestId('tab-2-pressable'))

    const nextTab = await screen.findByTestId('tab-2-text')

    expect(nextTab).toBeTruthy()

    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('uses a custom tab bar if provided', () => {
    const CustomTabBar = ({ navigationState }: RenderTabBarProps<unknown>) => (
      <Text>{`Custom Tab Bar - Current Index: ${navigationState.index}`}</Text>
    )

    render(<TabView tabs={tabs} CustomTabBar={CustomTabBar} />)

    // Check that the custom tab bar is rendered
    expect(screen.getByText('Custom Tab Bar - Current Index: 0')).toBeTruthy()
  })
})
