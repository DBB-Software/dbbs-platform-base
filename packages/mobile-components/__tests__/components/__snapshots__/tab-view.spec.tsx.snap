// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabView Component renders the correct content when the tab changes 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        collapsable={false}
        handlerTag={2}
        handlerType="NativeViewGestureHandler"
        horizontal={true}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        style={
          {
            "width": 750,
          }
        }
        waitFor={
          [
            {
              "current": null,
            },
          ]
        }
      >
        <View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                },
                {
                  "minWidth": 750,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "paddingLeft": 20,
                  "paddingVertical": 20,
                }
              }
              testID="tab-1-pressable"
            >
              <Text
                onLayout={[Function]}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    {
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                  ]
                }
                testID="tab-1-text"
              >
                Tab 1
              </Text>
              <View
                collapsable={false}
                jestAnimatedStyle={
                  {
                    "value": {
                      "backgroundColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 10,
                      "height": 2,
                      "width": 0,
                    },
                  }
                }
                style={
                  {
                    "backgroundColor": "rgba(103, 80, 164, 1)",
                    "borderRadius": 10,
                    "height": 2,
                    "width": 0,
                  }
                }
                testID="tab-1-animated"
              />
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "paddingLeft": 20,
                  "paddingVertical": 20,
                }
              }
              testID="tab-2-pressable"
            >
              <Text
                onLayout={[Function]}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    {
                      "fontWeight": "400",
                      "textAlign": "center",
                    },
                  ]
                }
                testID="tab-2-text"
              >
                Tab 2
              </Text>
              <View
                collapsable={false}
                jestAnimatedStyle={
                  {
                    "value": {
                      "backgroundColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 10,
                      "height": 2,
                      "width": 0,
                    },
                  }
                }
                style={
                  {
                    "backgroundColor": "rgba(103, 80, 164, 1)",
                    "borderRadius": 10,
                    "height": 2,
                    "width": 0,
                  }
                }
                testID="tab-2-animated"
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <LEGACY_RNCViewPager
    collapsable={false}
    initialPage={0}
    keyboardDismissMode="on-drag"
    layout={
      {
        "height": 0,
        "width": 0,
      }
    }
    layoutDirection="ltr"
    onMoveShouldSetResponderCapture={[Function]}
    onPageScroll={[Function]}
    onPageScrollStateChanged={[Function]}
    onPageSelected={[Function]}
    scrollEnabled={true}
    style={
      {
        "flex": 1,
      }
    }
    useLegacy={true}
  >
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "flex": 1,
              "overflow": "hidden",
            },
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            [
              undefined,
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ],
          ]
        }
      >
        <Text>
          Tab 1 Content
        </Text>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        style={
          [
            {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            [
              undefined,
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ],
          ]
        }
      />
    </View>
  </LEGACY_RNCViewPager>
</View>
`;
